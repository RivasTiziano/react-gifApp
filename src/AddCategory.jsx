import { useState } from "react";

export const AddCategory = ({onNewCategory}) => {

    const [addCategory, setAddCategory] = useState("");

    const handleChange = ({target}) => {
        setAddCategory(target.value.trim());
    };

    const handlePress = ({key, target}) => {
        if(key === 'Enter' && addCategory.length >= 1){
            onNewCategory(addCategory);
            setAddCategory("");
            target.value = "";
        };
    };

    return (
        <input 
            type="text"
            placeholder="Introduce your Category"
            onChange={ handleChange }
            onKeyPress={ handlePress }
        />
  );
};

