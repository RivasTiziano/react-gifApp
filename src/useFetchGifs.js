import { useEffect, useState } from "react";
import { getGifs } from "./getGifs";

export const useFetchGifs = (category) => {

    const [gifs, setGifs] = useState([]);
    const [loading, isLoading] = useState(true);

    useEffect(()=>{
        getGifs(category).then(r=>setGifs(r));
        isLoading(false);
    }, []);

  return {
    gifs,
    loading
  }
}
