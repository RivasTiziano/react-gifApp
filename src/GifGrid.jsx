import { GifItem } from "./GifItem";
import { useFetchGifs } from "./useFetchGifs";

export const GifGrid = ( {category} ) => {

  const {gifs, loading} = useFetchGifs(category);

  return (
    <>
        {loading && <h3>loading...</h3>}
        <h3>{ category }</h3>
        <div className="card-grid">
          { gifs.map((gifs)=><GifItem key={gifs.id} {...gifs}/>) }
        </div>
    </>
  )
}