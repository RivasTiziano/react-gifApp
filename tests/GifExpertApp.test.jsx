import { fireEvent,render,screen, waitFor } from "@testing-library/react"
import { GifExpertApp } from "../src/GifExpertApp"


describe('Tests in GifExpertApp', () => { 

    test('The snapshot should be approve', () => { 

        const {container} = render(<GifExpertApp/>)
        
        expect(container).toMatchSnapshot()

     })

    test('should create the <h3> node with the value', async () => { 

        render(<GifExpertApp/>)
        
        const value = "itachi";
        const inputElement = screen.getByRole("textbox")

        fireEvent.input( inputElement, {
                target: {
                    value
                }
            }
        )

        fireEvent.keyDown( inputElement, {key: 'Enter'} )

        expect(screen.getByText(value))

        
        await waitFor(()=>{
            expect(screen.getAllByRole("img").length).toBeGreaterThan(0);
        })

        screen.debug()
     })
 })