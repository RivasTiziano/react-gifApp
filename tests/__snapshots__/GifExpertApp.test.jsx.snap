// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in GifExpertApp The snapshot should be approve 1`] = `
<div>
  <h1>
     GifExpertApp 
  </h1>
  <input
    placeholder="Introduce your Category"
    type="text"
  />
</div>
`;
